cpp/src/io/json/json_column.cu

```diff
@@ -410,12 +410,31 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> device_json_co
       std::vector<std::unique_ptr<column>> child_columns;
       std::vector<column_name_info> column_names{};
       size_type num_rows{json_col.num_rows};
+      if (schema.has_value() and schema.value().column_order.has_value()) {
+        CUDF_EXPECTS(schema.value().child_types.size() == schema.value().column_order->size(),
+                     "Input schema column order size mismatch with input schema child types");
+      }
       // Create children columns
-      for (auto const& col_name : json_col.column_order) {
-        auto const& col = json_col.child_columns.find(col_name);
-        column_names.emplace_back(col->first);
-        auto& child_col           = col->second;
+      auto const& col_order = schema.has_value() and schema.value().column_order.has_value() and
+                                  not schema.value().column_order->empty()
+                                ? schema.value().column_order.value()
+                                : json_col.column_order;
+      for (auto const& col_name : col_order) {
         auto child_schema_element = get_child_schema(col_name);
+        auto const& found_col     = json_col.child_columns.find(col_name);
+        if (prune_columns and found_col == std::end(json_col.child_columns)) {
+          column_names.emplace_back(make_column_name_info(
+            child_schema_element.value_or(schema_element{data_type{type_id::EMPTY}}), col_name));
+          auto all_null_column = make_all_nulls_column(
+            child_schema_element.value_or(schema_element{data_type{type_id::EMPTY}}),
```

Comment:
Here we should have the schema available. If not, there should be something wrong and we should throw instead of using `EMPTY` type.

Reply:
I added CUDF_EXPECTS.